tools/trace.py

```diff
@@ -393,6 +393,13 @@ def _generate_field_assign(self, idx):
                 bpf_probe_read(&__data.v%d, sizeof(__data.v%d), (void *)%s);
         }
                 """ % (expr, idx, idx, expr)
+                if expr.startswith('*'):
+                    ptr_expr = expr[1:]
+                    return text + """
+        if (%s != 0) {
+            bpf_probe_read(&__data.v%d, sizeof(__data.v%d), (void *)%s);
+        }
```

Comment:
I think we do not need the `if` condition in the above. User has a dereference and we should just honor it. bpf_probe_read won't crash the kernel. This will be consistent with other rewriter changes related to
bpf_probe_read as we do not really check return value for bpf_probe_read.

Typically, rewriter should handle `*` expressions properly. The exception is for anonymous function parameters like arg1, arg2, etc, which are replaced with PT_REGS_PARM1, etc. where `*PT_REGS_PARM1(ctx)` won't work due to type mismatch. One way could be to only handle `arg1/arg2/arg3/arg4/arg5/arg/retval` here and leave others to be handled by rewriter. But I am okay to handle all cases of `*<...>` here for simplicity as done in the above.

Could you put an example in trace.py and trace_example.txt? For example, the command line for your example in this pull request?